sdk/core/System.ClientModel/src/Options/ResponseStatusClassifier.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+
+namespace System.ClientModel.Primitives
+{
+    internal class ResponseStatusClassifier : PipelineMessageClassifier
+    {
+        private BitVector640 _successCodes;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="ResponseStatusClassifier"/>.
+        /// </summary>
+        /// <param name="successStatusCodes">The status codes that this classifier
+        /// will consider not to be errors.</param>
+        public ResponseStatusClassifier(ReadOnlySpan<ushort> successStatusCodes)
+        {
+            _successCodes = new();
+
+            foreach (int statusCode in successStatusCodes)
+            {
+                AddClassifier(statusCode, isError: false);
+            }
+        }
+
+        public sealed override bool IsErrorResponse(PipelineMessage message)
+        {
+            if (message.Response is null)
+            {
+                throw new InvalidOperationException("Response is not set on message.");
```

Comment:
Should this be consistent with https://github.com/Azure/azure-sdk-for-net/pull/41239/files#r1456109052 ?

Reply:
Happy to make those consistent - thanks for catching!